source/uploader.ts

```diff
@@ -177,7 +177,14 @@ export class Uploader<TEntityTypeMap extends Record<string, any>> {
   /** Initiate upload. Promise is resolved once preflight is complete and upload started. */
   async start() {
     logger.debug("Upload starting", this.componentId);
-    await this.uploadPreflight();
+    try {
+      await this.uploadPreflight();
+    } catch (error) {
+      if (this.onError) {
+        this.onError(error as Error);
+      }
+      return;
```

Comment:
We should probably retain the previous behavior where start throw:ed if the preflight failed.

Reply:
```suggestion
      if (this.onError) {
        this.onError(error as Error);
      }
      throw error;
```
<img width="807" alt="image" src="https://github.com/ftrackhq/ftrack-javascript/assets/40965022/28690062-f852-463e-8b03-e39ec23821c0">

When I throw an error like above instead of return, I receive this error message instead of `notifyError`. Is there any way to prevent this?